### PR TITLE
add move_to_start and move_to_index

### LINKED ISSUE #8: Create move_to_start
Create `move_to_start` and more generally `move_to_index(opts)`.

`opts` should be a table with a key `index`. This is so that we can use the same opts table spec accross different api methods, in a similar way to `grapple.nvim`.

### PR DESCRIPTION
**Pull request by artem-streltsov**
Closes #8

### CODE REVIEW

File: lua/grapple-utils.lua [hunk 1]
```diff
@@ -22,6 +13,58 @@ local function get_current_index(tags, current_path)
...
+	end
+
+	local current_index = get_current_index(tags, current_path)
+	if not current_index then
+		vim.notify("Tag not found for the current buffer", vim.log.levels.ERROR)
+		return
+	end
+
+	if not opts or not opts.index then
+		vim.notify("Invalid options: index is required", vim.log.levels.ERROR)
```
Comment (artem-streltsov) [inline lua/grapple-utils.lua]: Is `ERROR` log level ok here?

File: lua/grapple-utils.lua [hunk 2]
```diff
@@ -39,8 +39,8 @@ function M.move_to_index(opts)
 	end
 
 	local new_index = opts.index
-	if new_index == -1 then
-		new_index = #tags
+	if new_index < 0 then
```
Comment (will-lynas) [inline lua/grapple-utils.lua:43]: If we used the built in modulo operator here we could support any index

File: lua/grapple-utils.lua (hunk 1 above)
Comment (will-lynas) [reply lua/grapple-utils.lua]: Yes. I think this should be at the very start of the function too.

File: lua/grapple-utils.lua [hunk 3]
```diff
@@ -22,6 +13,58 @@ local function get_current_index(tags, current_path)
 	return nil
 end
 
+function M.move_to_index(opts)
```
Comment (will-lynas) [inline lua/grapple-utils.lua:20]: This function is very similar to just using `require("grapple").tag({index = index})`.
The difference is that this function supports indices that `grapple` considers out of bounds using modulo.
I think the name should reflect this modulo behaviour.

File: lua/grapple-utils.lua [hunk 4]
```diff
@@ -33,16 +38,8 @@ function M.move_to_index(opts)
...
-		vim.notify("Invalid options: index is required", vim.log.levels.ERROR)
-		return
-	end
-
 	local new_index = opts.index
-	if new_index < 0 then
-		new_index = (#tags + 1) + new_index
-	end
-
+	new_index = new_index % #tags + 1
```
Comment (will-lynas) [inline lua/grapple-utils.lua]: Because of 1 based indexing this should be `(new_index - 1) % #tags + 1`

File: lua/grapple-utils.lua [hunk 5]
```diff
@@ -22,6 +13,55 @@ local function get_current_index(tags, current_path)
 	return nil
 end
 
+function M.cycle_to_index(opts)
```
Comment (will-lynas) [inline lua/grapple-utils.lua]: Actually I think just `move_to_index` was better. We can probably add a docstring describing the modulo behaviour later.

Comment (will-lynas) [reply lua/grapple-utils.lua]: Oh, actually 1-based indexing makes this a bit harder than just modulo.

File: lua/grapple-utils.lua [hunk 6]
```diff
@@ -22,6 +13,58 @@ local function get_current_index(tags, current_path)
...
+	local current_path = vim.api.nvim_buf_get_name(0)
+	local current_opts = { path = current_path }
+	if not grapple.exists(current_opts) then
+		vim.notify("Current buffer is not tagged", vim.log.levels.INFO)
+		return
+	end
+
+	local tags, err = grapple.tags()
+	if not tags then
+		vim.notify("No tags available: " .. err, vim.log.levels.ERROR)
```
Comment (will-lynas) [inline lua/grapple-utils.lua]: This should never happen, so should be an actual lua error or `assert` rather than a `vim.notify`.

**Review by will-lynas — APPROVED** [review]

**Pull request merged**